tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,306 @@
+if ($useConsent === "true") {
+    $updateValue = ['Value' => $useConsent];
+    $db->update(
+        'Config',
+        $updateValue,
+        array(
+         'ConfigID' => $configID, 
+        )
+    );
+    echo "\nConfig settings set to use consent.\n\n";
+}
+/////////////////////////////////////////////
+// START IMPORT OF CONSENT INTO NEW TABLES //
+/////////////////////////////////////////////
+
+$dataArray  = [];
+foreach ($consentList as $consentName=>$consentLabel) {
+    // Populate consent table with consents from Config.xml
+
+    // If consent doesn't already exist in `consent` table, insert
+    $exists = $db->pselectOne(
+        "SELECT * FROM consent WHERE Name=:consentname",
+        array('consentname' => $consentName)
+    );
+    if (empty($exists)) {
+        $db->insert(
+            'consent',
+            array(
+               'Name'  => $consentName,
+               'Label' => $consentLabel,
+             )
+        );
+    }
+    // Save ConsentID inserted
+    $consentID = $db->pselectOne(
+        "SELECT ConsentID FROM consent WHERE Name=:consentName",
+        array('consentName' => $consentName)
+    );
+    // Get all data where consent status has a value in participant_status
+    // Consent status without value means the candidate does not have this specific study consent
+    $consentDate       = $consentName . "_date";
+    $consentWithdrawal = $consentName . "_withdrawal";
+    $dataQuery         = "SELECT CandID, " . $db->escape($consentName) . ", " . $db->escape($consentDate) . ", " . $db->escape($consentWithdrawal) . " FROM participant_status 
+                         WHERE " . $db->escape($consentName) . " IS NOT NULL OR " . $db->escape($consentName) . " <> ''";
+    $psData            = $db->pselect($dataQuery, array());
+    foreach ($psData as $entry) {
+        // Format data
+        $consentValues = [
+            'CandidateID'   => $entry['CandID'],
+            'ConsentID'     => $consentID,
+            'Status'        => $entry[$consentName],
+            'DateGiven'     => $entry[$consentDate],
+            'DateWithdrawn' => $entry[$consentWithdrawal],
+        ];
+        // Push each formatted entry to array
+        array_push($dataArray, $consentValues);
+    }
+}
+// Output list of data to be inserted into new table
+echo "\nRows to be inserted into 'candidate_consent_rel' table ..\n";
+echo "Empty dates will be inserted as NULL.\n\n";
+print_r($dataArray);
+
+// Populate candidate_consent_rel
+foreach ($dataArray as $consentValues) {
+    $db->insert('candidate_consent_rel', $consentValues);
+}
+echo "\nConsent data insert complete.\n";
+
+// Select consent history and import into new history table
+$historyFieldsQuery = "SELECT CandID, entry_staff, data_entry_date";
+
+foreach ($consentList as $consentName=>$consentLabel) {
+    $historyFieldsQuery .= ", " . $db->escape($consentName) . ", " . $db->escape($consentName) . "_date, " . $db->escape($consentName) . "_withdrawal";
```

Comment:
@zaliqarosli this is your problem right here. this query ends up looking like this
```
SELECT CandID, entry_staff, data_entry_date, `study_consent_cnd`, `study_consent_cnd`_date, `study_consent_cnd`_withdrawal, `study_consent_T6`, `study_consent_T6`_date, `study_consent_T6`_withdrawal, `study_consent_T10`, `study_consent_T10`_date, `study_consent_T10`_withdrawal, `study_consent_ONDRI_CND`, `study_consent_ONDRI_CND`_date, `study_consent_ONDRI_CND`_withdrawal FROM consent_info_history
```

Notice the back tickets in the example below, they are in the wrong positions they are resulting in column fields called just `_date`... thus notfinding the expected data.
```
`study_consent_T6`_date
```

Reply:
ahh! that came up before and i thought i fixed it. thank you @ridz1208 !!!!